esp-hal/src/spi/master.rs

```diff
@@ -1174,13 +1223,27 @@ mod dma {
         }
     }
 
+    impl<M, T> SetConfig for SpiDma<'_, M, T>
+    where
+        T: Instance,
+        M: Mode,
+    {
+        type Config = Config;
+        type ConfigError = Infallible;
```

Comment:
This isn't infallible. I mean, it is currently, but it shouldn't be. For instance, setting the bus freq to `1_000.MHz()` should be an error. I'm not saying we have to tackle all the errors in this PR, but we should at least stub out an `Error` enum instead of marking it as infallible.

Reply:
Will do